iModelCore/Units/PublicAPI/Formatting/FormattingApi.h

```diff
@@ -370,6 +408,7 @@ friend struct Format;
 
         bool UpdateProblemCode(FormatProblemCode code) { return m_problem.UpdateProblemCode(code); }

```

Comment:
This is a very strange method. The second parameter returns a pointer to the first parameter, and the method always returns an empty Quantity when we probably have a quantity object with a more precise error value available. I don't like this method, and I don't mind the bit of redundancy which it fixes. Let's just replace the calls, for example

return UpdateAndSetProblemCode(FormatProblemCode::QT_MathematicOperationFoundButIsNotAllowed, probCode);

By the full bloated version, like

probCode = FormatProblemCode::QT_MathematicOperationFoundButIsNotAllowed;
return converted; //quantity has error assigned

And then at the outmost caller (GetQuantity()) we can check if a problemCode has been set to the pointer and if so, call m_problem.UpdateProblemCode(code);